### PR TITLE
Feature -Autocomplete city names feature

### PR DESCRIPTION
**Pull request by gbudjeakp**
Todos:
- [x] fetch cities from backend and render in flight search "to" and "from" inputs
- [x] Add Add debouncing

### CODE REVIEW

File: client/src/pages/FavoriteDestinantions.js [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+        //Assuming that the explore page can be accessed by both authenticated and unAuthenticated user.
+        const cityListResponse = await fetch("/api/cities", {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        const cityList = await cityListResponse.json();
+        cityList.map(city => {
+          city.favorite = favoriteList.indexOf(city.name) >= 0;
```
Comment (johnsciutto) [inline client/src/pages/FavoriteDestinantions.js]: `favoriteList` here is just an empty array, the value of `favoriteList.indexOf(city.name) >= 0` is always going to be `false`.

**Review by johnsciutto — COMMENTED** [review]
Looks good.
Remove the `console.log`(s) if they were just for checking/testing. If you're using them to deliberately display an error to the console, you could use `console.error()`, which is more specific for that purpose.

File: client/src/pages/FavoriteDestinantions.js (hunk 1 above)
Comment (gbudjeakp) [reply client/src/pages/FavoriteDestinantions.js]: Will effect the changes and corrections on next push.

File: client/src/pages/FavoriteDestinantions.js [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+        const cityListResponse = await fetch("/api/cities", {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        const cityList = await cityListResponse.json();
+        cityList.map(city => {
+          return city.name
+        });
```
Comment (johnsciutto) [inline client/src/pages/FavoriteDestinantions.js]: Check out the docs on the [`map` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#when_not_to_use_map). I don't think this is doing what you think it is 😉

**Review by johnsciutto — COMMENTED** [review]
Nice work 🙂

File: client/src/pages/FavoriteDestinantions.js (hunk 2 above)
Comment (gbudjeakp) [reply client/src/pages/FavoriteDestinantions.js]: Oh I see. I shouldn't even be mapping there at all. I've effected the change.

File: client/src/components/FlightSearch.js [hunk 3]
```diff
@@ -1,66 +1,78 @@
...
+           "Content-Type": "application/json",
+         },
+       });
+       const cityList = await cityListResponse.json();
+       setCities(cityList)
+      } catch(err){
+        console.error()
+      }
+    }
+    getCities()
```
Comment (sina-jamshidi) [inline client/src/components/FlightSearch.js]: We don't want to do this on component start. We are not using the pre-determined cities API that we were using on the explore page. That's just for a curated list of few cities. On this page we want the user to search for flights from anywhere in the world to anywhere else in the world. So we are using `/places/:regionId` which in turn hits up [this API](https://rapidapi.com/skyscanner/api/skyscanner-flight-search?endpoint=5a9c9edde4b084deb4ea6195).
Actually looking at `/places/:regionId` I don't even think `:regionId` is necessary so you can probably remove it

File: client/src/pages/FavoriteDestinantions.js [hunk 4]
```diff
@@ -1,11 +1,34 @@
...
+          },
+        });
+        const cityList = await cityListResponse.json();
+        setPlaces(cityList);
+      } catch (error) {
+        console.error();
+      }
+    }
+    getData();
+  }, []);
```
Comment (sina-jamshidi) [inline client/src/pages/FavoriteDestinantions.js]: this component is not part of this feature so we shouldn't be editing it here

**Review by sina-jamshidi — CHANGES_REQUESTED** [review]

File: client/src/components/FlightSearch.js (hunk 3 above)
Comment (gbudjeakp) [reply client/src/components/FlightSearch.js]: Oh I see. I'll effect the changes now.

**Review by sina-jamshidi — COMMENTED** [review]
Looks good but still no debouncing right?

File: client/src/pages/FavoriteDestinantions.js (hunk 4 above)
Comment (gbudjeakp) [reply client/src/pages/FavoriteDestinantions.js]: Held of on adding anything cause I wanted to make sure my code matched with Shays before doing anything.

**Review by sina-jamshidi — APPROVED** [review]

**Pull request merged**